modules/code-editor/src/migrations/v12_1_0-1565637584-restrict_rights_on_global_files.ts

```diff
@@ -0,0 +1,73 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+
+const failureMigration: sdk.MigrationResult = { success: false, message: 'workspaces.json has invalid format' }
+
+const isMigrationAlreadyDone = (rules: any[]) => {
+  const ressources: string[] = rules.map(r => r.res)
+  const alreadySpecifiedPermission = ressources.find(r => r.startsWith('module.code-editor.global'))
+  return !!alreadySpecifiedPermission
+}
+
+const migration: sdk.ModuleMigration = {
+  info: {
+    description: '',
+    target: 'core',
+    type: 'config'
+  },
+  up: async ({ bp }: sdk.ModuleMigrationOpts): Promise<sdk.MigrationResult> => {
+    const ghost = bp.ghost.forGlobal()
+
+    if (!ghost.fileExists('config', 'code-editor.json')) {
+      return failureMigration
+    }
```

Comment:
This could be possible if the user wasn't using the module before. 

Also, instead of all those validations (isArray, etc), you could just wrap that in a try catch and assume that the format is correct